pages/battle.php

```diff
@@ -235,18 +237,19 @@ function processBattleFightEnd(BattleManager|BattleManagerV2 $battle, User $play
         $player->pvp_immunity_ms = System::currentTimeMs() + (5 * 60 * 1000); // 5 minutes
 
         if ($battle->is_retreat) {
+            // Set health, move to village, and handle reputation loss (if any)
             $player->health = 5;
             $player->moveToVillage();
-            // Calc rep loss (if any)
             if (UserReputation::PVP_REP_ENABLED) {
                 $rep_lost = $player->reputation->handlePvPLoss($player, $battle->opponent, true);
                 if ($rep_lost > 0) {
                     $result .= "You have lost $rep_lost village reputation.[br]";
                 }
-                // Loot - winner takes half loser's if retreat
-                $system->db->query("UPDATE `loot` SET `user_id` = {$player->user_id}, `battle_id` = NULL WHERE `battle_id` = {$player->battle_id} LIMIT " . ceil($battle->getTotalLoot() / 2));
+                // Winner takes half of loser's loot if retreat
+                $half_loot_lost = ceil($battle->getTotalLoot() / 2);
+                $system->db->query("UPDATE `loot` SET `user_id` = {$battle->opponent->user_id}, `battle_id` = NULL WHERE `battle_id` = {$player->battle_id} LIMIT $half_loot_lost");
```

Comment:
Per above, as this runs first we can use this to let the retreating player claim half of their own loot
```suggestion
                // Keep half of own loot, the rest will default to the winner
                $system->db->query("
                  UPDATE `loot` SET `battle_id` = NULL 
                  WHERE `battle_id` = {$player->battle_id} AND `user_id`={$player->user_id} 
                  LIMIT  (SELECT COUNT(*) FROM `loot` WHERE `user_id`={$player->user_id}) * 0.5");
```